### PR TITLE
Autogenerate the monster switch statement

### PR DESCRIPTION
**Pull request by ejcx**
I have no clue how to test this besides just calling lots of API commands and verifying they are working. I also have no clue how to _write_ tests to maintain the autogenerated code and the autogeneration program. There's some obvious stuff to do, but beyond that Im not super sure.

This PR moves the switch and business logic from cmd/root.go to an autogenerated one in cmd/autogenerated.go, and all the functions that call cloudflare-go that are hand written are in the newly created cmd/api.go file.

Why? I think it's a little cleaner and makes the project slightly easier to contribute to but it doesn't solve any direct problem. It means contributors have one fewer place to add code.

### CODE REVIEW

**Comment by ejcx** [comment]
@abraithwaite since you have good and strong opinions, I'd love to hear your opinions about how to test my horrible spaghetti code in definitions.go

**Comment by ejcx** [comment]
@elithrar I added you to review but please ignore until you're free. I know how busy you are the next few weeks

File: definitions/definitions.go [hunk 1]
```diff
@@ -25,7 +26,20 @@ var (
 func init() {
 {{ .Commands }}
 }
+
+func Run(cmd *cobra.Command, args []string, name string, api *cloudflare.API) (resp interface{}, err error) {
```
Comment (abraithwaite) [inline definitions/definitions.go:30]: This is minor, but in Go it's idiomatic to put what's most important at the top of the file.  I think it'd be a bit cleaner to move these to the bottom of the file.

**Review by abraithwaite — COMMENTED** [review]
I looked through `definitions.go` and didn't see anything majorly wrong with it.  This program is really just doing a bunch of plumbing, so I'd say testing the autogenerated stuff is less valuable than if you're adding a ton of logic around the calls.

Generally, for something like this I'd unit test as much as I can but leave the rest untested.  It doesn't make sense to mock out the api given it's surface area.

One type of test I have seen in similar autogenerated scenarios is testing against a smaller, well known definition for it's behavior.  e.g. make a definition.toml with one single command, generate the code, point the api call at localhost, call the binary with different args, verify that the received call (body, headers) is what you expect it to be.

**Pull request merged**